winter_messaging_transactional/tests/tests/test_consumer.py

```diff
@@ -0,0 +1,261 @@
+
+    # Assert
+    inbox_messages = read_all_inbox_messages(session)
+    assert len(inbox_messages) == 1
+
+    event_message = inbox_messages[0]
+    assert event_message['consumer_id'] == 'consumer_timeout'
+    assert event_message['name'] == 'RetryableEvent'
+    assert event_message['counter'] == 0
+    assert event_message['received_at']
+    if can_be_handled_on_retry:
+        assert event_message['processed_at']
+        assert timeout_logs_count == 1
+    else:
+        assert event_message['processed_at'] is None
+        assert timeout_logs_count == 2
+
+    consumer_dao = injector.get(ConsumerDAO)
+    consumed_event = consumer_dao.find_by_id(event_id)
+
+    if can_be_handled_on_retry:
+        assert consumed_event
+        assert consumed_event['id'] == event_id
+        assert consumed_event['payload'] == payload
+    else:
+        assert consumed_event is None
+
+
+def test_consume_interrupt_during_timeout(database_url, rabbit_url, event_processor, injector, session):
+    event_publisher = injector.get(OutboxEventPublisher)
+    event_id = 1
+    payload = 'consumer_timeout'
+
+    # Act
+    process = run_consumer(database_url=database_url, rabbit_url=rabbit_url, consumer_id='consumer_timeout')
+    event = RetryableEvent(id=event_id, payload=payload)
+    event_publisher.emit(event)
+    session.commit()
+    sleep(EVENT_HANDLING_TIMEOUT // 2)
+
+    # Worker should not attempt to handle the event again after the TimeoutException if it has received the INT signal.
+    process.terminate()
+    sleep(EVENT_HANDLING_TIMEOUT)
+
+    # Assert
+    output = process.stdout.read1().decode('utf-8')
+    timeout_error_message = "WARNING:event_handling:Timeout is raised during execution _dispatch_event with args: ()," \
+                            " kwargs: {'event': RetryableEvent(id=1, payload='consumer_timeout'"
+
+    # Because there should not be a second attempt to handle the event, we expect only one timeout message in the logs
+    assert output.count(timeout_error_message) == 1
+
+    inbox_messages = read_all_inbox_messages(session)
+    assert len(inbox_messages) == 1
+
+    event_message = inbox_messages[0]
+    assert event_message['consumer_id'] == 'consumer_timeout'
+    assert event_message['name'] == 'RetryableEvent'
+    assert event_message['counter'] == 0
+    assert event_message['received_at']
+    assert event_message['processed_at'] is None
+
+    consumer_dao = injector.get(ConsumerDAO)
+    consumed_event = consumer_dao.find_by_id(event_id)
+    assert consumed_event is None
+
+
+@pytest.mark.parametrize('can_be_handled_on_retry', (True, False))
+def test_consume_with_error(database_url, rabbit_url, event_processor, injector, session, can_be_handled_on_retry):
+    event_publisher = injector.get(OutboxEventPublisher)
+    event_id = 1
+    payload = 'consume_with_error'
+
+    # Act
+    with event_consumer(database_url, rabbit_url, consumber_id='consumer_with_error'):
```

Comment:
nit: you can hide some details of instantiation process if create fixture for this that remember part of args like: database_url and rabbit_url and return callable(context manager) that accept only consumber_id.
for example using partial